src/components/ZoomableImage.vue

```diff
@@ -0,0 +1,90 @@
+<script setup>
+import { defineProps, ref} from 'vue'
+// eslint-disable-next-line no-unused-vars
+const props = defineProps(['imageSrc'])
+const image = ref(null)
+const imageLeftPos = ref(0)
+const imageTopPos = ref(0)
+const currentZoom = ref(1)
+
+const MIN_ZOOM = 1
+const MAX_ZOOM = 3
+const STEP_SIZE = 0.1
+let mouseX
+let mouseY
+
+const zoom = (e, direction) => {
+	if (!direction) direction = e.deltaY > 0 ? -1 : 1
+	let newZoom = currentZoom.value + direction * STEP_SIZE 
+	if (newZoom > MIN_ZOOM && newZoom < MAX_ZOOM) currentZoom.value = newZoom
+}
+
+// save the position clicked and add the pointermove listener
+const pointerDown = (e) => {
+	mouseX = e.pageX
+	mouseY = e.pageY
+	image.value.onpointermove = pointerMove
+}
+
+// use mouse movement delta to set position, limit so image stays on screen
+const pointerMove = (e) => {
+	let x 						= imageLeftPos.value+e.pageX-mouseX
```

Comment:
not sue if this is just github being weird but I see something like this:
`let x                                                            = imageLeftPos. etc`
and same for everything in this function

Reply:
I did that to line up all the math and make it more readable, but I can change that if its not helpful then